homeassistant/components/glances/sensor.py

```diff
@@ -218,6 +218,9 @@ async def async_update(self):  # noqa: C901
             except KeyError:
                 self._state = STATE_UNAVAILABLE
         elif self.entity_description.type == "raid":
-            for raid_device, raid in value["raid"].items():
-                if raid_device == self._sensor_name_prefix:
-                    self._state = raid[self.entity_description.key]
+            try:
+                self._state = value["raid"][self._sensor_name_prefix][
+                    self.entity_description.key
+                ]
+            except KeyError:
+                self.state = STATE_UNAVAILABLE
```

Comment:
Don't set unavailable state directly. Please set the `available` property to `False` instead.

Reply:
The Docker sensors of glances also sets the state to unavailable. I think it should be done consistent.

Reply:
Yes, please change this so it's consistent.